### ISSUE TITLE
Add support for multiple(array of) environment-annotations

### ISSUE THREAD

**Issue by Ashok-Varma** [Milad-Akarie/injectable#91]
Currently if we want to change dependency for one environment, we need to specify dependencies for every environment separately in different methods even if they are same
```
  @prod
  @lazySingleton
  CloudFunctions cloudFunctions() => CloudFunctions.instance;

  @pref
  @lazySingleton
  CloudFunctions cloudFunctions() => CloudFunctions.instance;

  @dev
  @lazySingleton
  CloudFunctions cloudFunctions() => CloudFunctions.instance;

  @test
  @lazySingleton
  CloudFunctions devCloudFunctions() => CloudFunctions.instance
      .useFunctionsEmulator(origin: "http://localhost:5001");
```

Instead
1. if we can annotate multiple environments for same dependency. 
```
  @prod
  @pref
  @test
  @lazySingleton
  CloudFunctions cloudFunctions() => CloudFunctions.instance;

  @dev
  @lazySingleton
  CloudFunctions devCloudFunctions() => CloudFunctions.instance
      .useFunctionsEmulator(origin: "http://localhost:5001");
```
Similar for 
```
  @LazySingleton(Env = {prod, pref, test})
  CloudFunctions cloudFunctions() => CloudFunctions.instance;
```

or
2. we can take the non-env specified dep as default fr all
```
  // this will be used for all other ENV's other than dev.
  @lazySingleton
  CloudFunctions cloudFunctions() => CloudFunctions.instance;

  @dev
  @lazySingleton
  CloudFunctions devCloudFunctions() => CloudFunctions.instance
      .useFunctionsEmulator(origin: "http://localhost:5001");
```

**Comment by britannio** [comment]
This looks like a duplicate of https://github.com/Milad-Akarie/injectable/issues/30 but I'd appreciate syntax like `@LazySingleton(environments: {'prod', 'pref', 'test'})` as sometimes I want to inject a class that's only supported on a certain platform E.g. MacOS but not Windows as is the case with a lot of the Firebase packages.

**Comment by Milad-Akarie** [comment]
finally implemented in injectable: 1.0.0 check it out!
